src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2418,6 +2418,37 @@ private RuntimeTypeCache InitializeCache()
             return cache;
         }
 
+        internal void ClearCache()
+        {
+            // If there isn't a GCHandle yet, there's nothing more to do.
+            if (Volatile.Read(ref m_cache) == IntPtr.Zero)
+            {
+                return;
+            }
+
+            // Loop until the cache is successfully zero'd out.
+            do
+            {
+                // If the GCHandle doesn't wrap a cache yet, there's nothing more to do.
+                RuntimeTypeCache? existingCache = (RuntimeTypeCache?)GCHandle.InternalGet(m_cache);
+                if (existingCache is null)
+                {
+                    return;
+                }
+
+                // Create a new, empty cache to replace the old one and try to substitute it in.
+                var newCache = new RuntimeTypeCache(this);
+                if (ReferenceEquals(GCHandle.InternalCompareExchange(m_cache, newCache, existingCache), existingCache))
```

Comment:
It should be enough to just set the GCHandle target to null. The GCHandle is weakhandle and so the code should be generally prepared to handle the null case.

#51070